### PR TITLE
configured Spring Security with DB authorization

### PR DESCRIPTION
**Pull request by KseniiaMakarova**

### CODE REVIEW

File: src/main/java/mateacademy/amazonreviewsapp/security/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    @PostMapping("/login")
+    public ResponseEntity<Object> login(@RequestBody @Valid UserLoginDto userLoginDto) {
+        User user = authenticationService.login(
+                userLoginDto.getProfileName(), userLoginDto.getPassword());
+        String token = jwtTokenProvider.createToken(user.getProfileName(),
+                user.getRoles().stream()
+                        .map(role -> role.getName().name())
+                        .collect(Collectors.toList()));
+        Map<String, Object> body = new HashMap<>();
```
Comment (boroda4436) [inline src/main/java/mateacademy/amazonreviewsapp/security/AuthenticationController.java]: В якості body я б можливо використав щось типу https://www.javatips.net/api/extdirectspring-master/src/main/java/ch/ralscha/extdirectspring/bean/BaseResponse.java але параметризований дженериком. Наприклад: BaseResponse<UserResponseDto> responseBody = new BaseResponse<>();

Comment (KseniiaMakarova) [reply src/main/java/mateacademy/amazonreviewsapp/security/AuthenticationController.java]: @boroda4436 I suppose I may use the following code here:
`BaseResponse<String> responseBody = new BaseResponse<>();`
`responseBody.setType(ResponseType.EVENT); // we may have an enum {EVENT, EXCEPTION} specified in BaseResponse class`
`responseBody.setValue(token); // value field is of type T`

what do you think?
also, can we use BaseResponse in GlobalExceptionHandler as well?

Comment (boroda4436) [reply src/main/java/mateacademy/amazonreviewsapp/security/AuthenticationController.java]: LGTM. But you can avoid responseBody.setType(ResponseType.EVENT);. Basically you can determine the type of response from the response status.
> can we use BaseResponse in GlobalExceptionHandler as well?

Yes, we should.

Comment (KseniiaMakarova) [reply src/main/java/mateacademy/amazonreviewsapp/security/AuthenticationController.java]: thanks!)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**